### PR TITLE
Support state param in the user authenticate call

### LINKED ISSUE #3: Support state
When opening the first GitHub link to let the user authenticate, one can add a `state` parameter:

> state: An unguessable random string. It is used to protect against cross-site request forgery attacks.

This `state` will then be sent back by GitHub on the second step:

> If the user accepts your request, GitHub redirects back to your site with a temporary code in a code parameter as well as the state you provided in the previous step in a state parameter. If the states don't match, the request has been created by a third party and the process should be aborted.

The issue is that the microservice has no chance of knowing the `state` users send to GitHub in the first step.

----

One way I figured out to work around this is to have people login by providing a link to

```
my-url.now.sh/login
```

Which would redirect to the GitHub login page with some random string stored in memory, which we can then compare in the second step.

Does that make sense?

*([GitHub source](https://developer.github.com/v3/oauth/#1-redirect-users-to-request-github-access))*

### PR DESCRIPTION
**Pull request by timReynolds**
This PR fixes #3 

I came to the same conclusion as the @mxstbr that the best way to do this is adding an additional url for login that populates the state param then redirects the user to the GitHub authorise page. 

As I've had to update the routes this is a breaking change. 

For simplicity a list of known states are stored in memory.

### CODE REVIEW

**Review by mxstbr — COMMENTED** [review]
Oh wow this looks amazing! Sadly I don't have time thoroughly review this (as you evidently see by my time to response, sorry...), is it finished?

**Comment by timReynolds** [comment]
@mxstbr functionally yes, ideally it would have some tests but given a test setup isn't present I didn't create one. 

This PR includes an updated readme so it should be easy to test manually. 

Things I consider before merging;

  - It's a breaking change, you have to use /login
  - Adds new dependencies 
  - Doesn't consider state being persisted, it's just an in memory array

**Comment by mxstbr** [comment]
Let's add a CHANGELOG.md and use pseudo-semver (make this `v1.0.0` maybe?) so people who want to update can check what changed and if it breaks them?

**Comment by timReynolds** [comment]
Sounds good, I'll take a look at adding this when I get 10.

**Comment by mxstbr** [comment]
Awesome, can't wait to land this!

**Comment by timReynolds** [comment]
Change log is now included which documents the 1.0.0 state change.

File: README.md [hunk 1]
```diff
@@ -34,11 +34,11 @@ When authentication was successful, the user will be redirected to the `REDIRECT
 
 ### Finish setup
 
-To make this work you have to set the authorization callback URL of [your application on GitHub](https://github.com/settings/developers) to whatever URL `now` gave you:
+To make this work you have to set the authorization callback URL of [your application on GitHub](https://github.com/settings/developers) to whatever URL `now` gave you plus the path `/callback` e.g. `http://localhost:3000/callback`:
 
 ![Authorization callback URL: 'your-url.now.sh'](https://cloud.githubusercontent.com/assets/7525670/22621592/95546272-eb27-11e6-80f3-6a2cd556d319.png)
```
Comment (mxstbr) [inline README.md]: Do we need to update this screenshot?

**Review by mxstbr — APPROVED** [review]
This is amazing dude, great job! :clap:

**Pull request merged**